pkg/rewards/2_goldStakerRewardAmounts.go

```diff
@@ -107,26 +108,29 @@ staker_operator_total_tokens AS (
     END as total_staker_operator_payout
   FROM staker_proportion
 ),
--- Calculate the token breakdown for each (staker, operator) pair
+-- Calculate the token breakdown for each (staker, operator) pair with dynamic split logic
+-- If no split is found, default to 1000 (10%)
 token_breakdowns AS (
-  SELECT *,
+  SELECT sott.*,
     CASE
-      WHEN snapshot < @amazonHardforkDate AND reward_submission_date < @amazonHardforkDate THEN
-        cast(total_staker_operator_payout * 0.10 AS DECIMAL(38,0))
-      WHEN snapshot < @nileHardforkDate AND reward_submission_date < @nileHardforkDate THEN
-        (total_staker_operator_payout * 0.10)::text::decimal(38,0)
+      WHEN sott.snapshot < @amazonHardforkDate AND sott.reward_submission_date < @amazonHardforkDate THEN
+        cast(sott.total_staker_operator_payout * COALESCE(oas.split, 1000) / 10000.0 AS DECIMAL(38,0))
```

Comment:
I would change this in one of two ways:

1. Remove the decimal point from `10000.0` and let postgres have its way with the type casting
2. Explicitly cast `10000` to a known floating point type (probably `decimal`)

when I run `select pg_typeof(10000.0)` postgres treats it as numeric, which is likely what we want, but we should probably make that explicit for the sake of consistency (decimal is just an alias to numeric)

Reply:
Actually, we shouldnt need to touch this line at all. Same with the nile hardfork. None of these calculations are relevant any more and occurred in the past pre rewards v2, so we should leave them alone to prevent a regression.

cc: @ypatil12

Reply:
Yup, agreed - much simpler!

Reply:
Done in [5c7f58c](https://github.com/Layr-Labs/sidecar/pull/117/commits/5c7f58cda135e3567ba668c9a5c152d780d59e4a)